simple/functions/server/serverfunction.ts

```diff
@@ -0,0 +1,136 @@
+        example: "/simple/serverfunction/echo?phrase=response  =>  {success: true, phrase: 'response'}",
+        method: "get"
+    },
+    echo: async (
+        _args: { phrase: string },
+        _content: any,
+        _serv: any,
+        _role: YRole,
+        _session: YSession
+    ): ServerFunctionResponse<{
+        success: boolean,
+        phrase: string
+    }> => {
+        return {data: {success: true, phrase: _args.phrase}};
+    },
+
+    $query: {
+        module: "simple",
+        name: "query",
+        scope: "public",
+        description: "returns calculated property of a model from the DB",
+        restricted: false,
+        args: [
+            {
+                name: "title",
+                type: "string",
+                mandatory: true,
+                description: "title of the model to search by"
+            }
+        ],
+        content: [],
+        output: "{success: true, content: 'qwe(asd)'}",
+        example: "/simple/serverfunction/query?title=qwe  =>  {success: true, content: 'qwe (asd)'}",
+        method: "get"
+    },
+    query: async (
+        _args: { title: string },
+        _content: any,
+        _serv: any,
+        _role: YRole,
+        _session: YSession
+    ): ServerFunctionResponse<{
+        success: boolean,
+        content: string
+    }> => {
+        const cnx = await _session.dbserver.connect("simple", 'simple', null);
+        const query = YQuery.From('simple.model')
+                    .Where(`obj.title == "${_args.title}"`)
+                    .Select()
+                    .Execute();
+        const result = await cnx.find(query);
+        const firstMatch:Model = await _role.simple.createClass("model");
+        firstMatch.setData(await result.iterator.next(null));
+        return {data: {success: true, content: firstMatch.cGetContent()}};
+    },
+
+    $add: {
+        module: "simple",
+        name: "add",
+        scope: "public",
+        description: "add a new model with given title and description",
+        restricted: false,
+        args: [],
+        content: [
+            {
+                name: "title",
+                type: "string",
+                mandatory: true,
+                description: "title of the model"
+            },
+            {
+                name: "description",
+                type: "string",
+                mandatory: true,
+                description: "description of the model"
+            }
+
+        ],
+        output: "{success: true}",
+        example: "/simple/serverfunction/add",
+        method: "post"
+    },
+    add: async (
+        _args: any,
+        _content: { title: string, description: string },
+        _serv: any,
+        _role: YRole,
+        _session: YSession
+    ): ServerFunctionResponse<{
+        success: boolean
+    }> => {
+        const newModel:Model = await _role.simple.createClass("model");
+        // .setData doesn't work
+        // newModel.setData(_content);
```

Comment:
from reading the `YClass.setData()` and the `YDBConnection.commit()` code it makes sense that it doesn't work — but in arpege-web there's code setting the state of objects in this way and then commit...  :(

Reply:
Yes, we do it ourself in public api. My understanding is that it is more hackish than intended use of setData, hence the comment. I might be wrong tho. Opinions are welcome.